scripts/pipeline/appropriate_ref_collection_fasta.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Classifies the appropriate set of comparators for a taxon by taxonomic-directory
+
+Example::
+
+    $ python appropriate_ref_collection_fasta.py --taxon_name canis_lupis_familiaris \
+        --genome_fasta ${ENSEMBLFTP}/Canis_lupus_familiaris/Canis_lupus_familiaris.fasta \
+        --url mysql://ensro@mysql-ens-sta-1:4519/ncbi_taxonomy_106 \
+        --taxon_name canis_lupis_familiaris \
+        --comparators_dir /hps/nobackup/flicek/ensembl/compara/shared/references_symlink_fasta \
+        --dest_dir '45.88.81.155:/bucket3'
+
+"""
+
+import os
+from os.path import basename
+from pathlib import Path
+from shutil import copytree, copyfile
+import sys
+
+from argparse import ArgumentParser
+
+from sqlalchemy.orm import Session
+
+from ensembl.database import DBConnection
+from ensembl.compara.utils.taxonomy import (
+    collect_taxonomys_from_path,
+    match_taxon_to_reference,
+)
+
+
+def _get_collection_path(session: Session, taxon: str, ref_dir: Path) -> Path:
```

Comment:
the functions starting by underscore is a convention used in python to denominate private methods.  I think we need an harmonisation to define if we need to use this convention in function defined in script or in class definition only. So far in the different scripts I reviewed I did not see this for scripts. But happy to get opinion form other team members.